wordpress-mobile/WordPress-iOS
# Adds support to share a site (blog) via Reader

### Description

Proposes a fix for #10898, which requests the ability to share a site (i.e., blog) via the reader. At the moment, users can share a post via the iOS "share sheet".

This change required some consideration for the fact that `ReaderStreamViewController` is used for heterogenous topic types. It leverages the existing `PostSharingController` and captures a new analytics event.

While the code can be reviewed as-is, it cannot be merged until we've secured approval from design (label affixed) and merged a [supporting PR](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/174).

To test:

- Checkout the branch and confirm that existing tests pass.
- Review the code change, taking note of the new analytics event captured.
- Open the Reader and navigate to a selection of topics, sites, and posts and observe the behavior.

Update release notes:


### Design Review

I took the liberty to expose a similar experience to the one used to share a blog post in the Reader detail screen. The animations below show the feature working in action with the WP share extension.

Some key distinctions:

- We use the "iOS share" Gridicons.
- We pass the following site metadata to the iOS "share sheet" : site title, description & URL. Its usage varies based on the share destination.

iPhone SE - Portrait - Site :_
![10898_iphone](https://user-images.githubusercontent.com/221062/51719152-318caa80-1ffd-11e9-943c-8f70822ade80.gif)

_iPad - Landscape - Discover :_
![10898_ipad-landscape](https://user-images.githubusercontent.com/221062/51719158-36515e80-1ffd-11e9-811c-b510d29aaab4.gif)
